addons/account/models/account_payment.py

```diff
@@ -963,8 +963,9 @@ def action_post(self):
                     method_name=self.payment_method_line_id.name,
                     partner=payment.partner_id.display_name,
                 ))
-
-        self.state = 'in_process'
+            # Avoid going back one state when clicking on the confirm action in the payment list view and having paid expenses selected
+            # We need to set values to each payment to avoid recomputation later
+            payment.state = 'in_process' if payment.state == 'draft' else payment.state
```

Comment:
Why not 
```suggestion
            # Avoid going back one state when clicking on the confirm action in the payment list view and having paid expenses selected
            # We need to set values to each payment to avoid recomputation later
            if payment.state == 'draft':
                payment.state = 'in_process' 
```
 ? :thinking: 


You could as well just do the following outside of the loop. 
self.filtered(lambda x: x.state == 'draft').state = 'in_process'

Not sure you need the comments either ; it's pretty straghtforward

Reply:
# We need to set values to each payment to avoid recomputation later
No, I tried I swear :p

Reply:
> > We need to set values to each payment to avoid recomputation later
> 
> No, I tried I swear :p

That does not explain why, nor which recomputation you're talking about. And it still feels strange :p

Reply:
because before my change, it would override all records
If I only override some records, when calling action_post in sale for example, it will recompute some of the dependency fields along the way, forcing a recompute of these. Switching them to 'draft' state instead of their original ones